pallet/sylo/src/lib.rs

```diff
@@ -0,0 +1,459 @@
+			tags: Option<Vec<Vec<u8>>>,
+		},
+		ValidationRecordDeleted {
+			author: T::AccountId,
+			id: Vec<u8>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight({
+			T::WeightInfo::set_payment_asset()
+		})]
+		pub fn set_payment_asset(origin: OriginFor<T>, payment_asset: AssetId) -> DispatchResult {
+			ensure_root(origin)?;
+			<SyloAssetId<T>>::put(payment_asset);
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight({
+			T::WeightInfo::set_sylo_resolver_method()
+		})]
+		pub fn set_sylo_resolver_method(
+			origin: OriginFor<T>,
+			resolver_method: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			<SyloResolverMethod<T>>::put(resolver_method);
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight({
+			T::WeightInfo::register_resolver(<T::StringLimit>::get(), <T::MaxServiceEndpoints>::get())
+		})]
+		pub fn register_resolver(
+			origin: OriginFor<T>,
+			identifier: BoundedVec<u8, T::StringLimit>,
+			service_endpoints: BoundedVec<ServiceEndpoint<T::StringLimit>, T::MaxServiceEndpoints>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				<Resolvers<T>>::get(identifier.clone()).is_none(),
+				Error::<T>::ResolverAlreadyRegistered
+			);
+
+			let resolver =
+				Resolver { controller: who.clone(), service_endpoints: service_endpoints.clone() };
+
+			<Resolvers<T>>::insert(identifier.clone(), resolver);
+
+			Self::deposit_event(Event::ResolverRegistered {
+				id: identifier.to_vec(),
+				controller: who,
+				service_endpoints,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight({
+			T::WeightInfo::update_resolver(<T::MaxServiceEndpoints>::get())
+		})]
+		pub fn update_resolver(
+			origin: OriginFor<T>,
+			identifier: BoundedVec<u8, T::StringLimit>,
+			service_endpoints: BoundedVec<ServiceEndpoint<T::StringLimit>, T::MaxServiceEndpoints>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let mut resolver =
+				<Resolvers<T>>::get(identifier.clone()).ok_or(Error::<T>::ResolverNotRegistered)?;
+
+			ensure!(who == resolver.controller, Error::<T>::NotController);
+
+			resolver.service_endpoints = service_endpoints.clone();
+
+			<Resolvers<T>>::insert(identifier.clone(), resolver);
+
+			Self::deposit_event(Event::ResolverUpdated {
+				id: identifier.to_vec(),
+				controller: who,
+				service_endpoints,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight({
+			T::WeightInfo::unregister_resolver()
+		})]
+		pub fn unregister_resolver(
+			origin: OriginFor<T>,
+			identifier: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let resolver =
+				<Resolvers<T>>::get(identifier.clone()).ok_or(Error::<T>::ResolverNotRegistered)?;
+
+			ensure!(who == resolver.controller, Error::<T>::NotController);
+
+			<Resolvers<T>>::remove(identifier.clone());
```

Comment:
nit: unnecessary clone (memory allocations) can be removed i think
```suggestion
			let resolver = <Resolvers<T>>::get(&identifier).ok_or(Error::<T>::ResolverNotRegistered)?;
            ensure!(who == resolver.controller, Error::<T>::NotController);
            
            <Resolvers<T>>::remove(&identifier);
```